test/profile/index.test.js

```diff
@@ -0,0 +1,42 @@
+
+const test = require('ava')
+const sinon = require('sinon')
+const os = require('os')
+const Profile = require('../../src/profile')
+
+test.beforeEach((t) => {
+  const config = {
+    getApiHost: () => 'http://localhost',
+    getApiKey: () => 'api_key'
+  }
+  const runlog = {
+    error: sinon.stub(),
+    debug: sinon.stub(),
+    record: sinon.stub(),
+    keys: {}
+  }
+  sinon.stub(os, 'homedir').returns('/Users/tester')
+  t.context.profile = new Profile({ config, runlog })
+})
+
+test('writes to all shell and power profiles for system with all shells and power available', async (t) => {
+  const { profile } = t.context
+  profile._isRunnable = sinon.stub().returns(true)
+
+  const {
+    detectedShellFormatProfiles,
+    detectedPowerFormatProfiles
+  } = await profile._detectProfiles()
+  t.deepEqual(detectedShellFormatProfiles, [
```

Comment:
so... this passed on the windows CI, which is somewhat surprising. but i think having these unix-style paths directly in the unit tests might give us some false sense of safety.

i think the point of this test is to make sure that all the shell files we know about are "detected" if all shells are runnable. if that's the intent, perhaps an easier way would be to stub `inHome` and `inConfig` to passthrough their args and assert a list of pathless profiles: `[.profile, .kshrc., ...]` ?